ui-components/src/actions/FavoriteButton.tsx

```diff
@@ -34,12 +31,24 @@ const FavoriteButton = ({ id, name }: FavoriteButtonProps) => {
     if (!profile || listingFavorited) {
       return
     }
-    preferences.favoriteIds?.push(id)
+    if (profile && !profile.preferences?.favoriteIds) {
+      if (!profile.preferences) {
+        profile.preferences = { favoriteIds: [id] }
+      } else if (!profile.preferences?.favoriteIds) {
+        profile.preferences.favoriteIds = [id]
+      }
+    } else {
+      preferences?.favoriteIds?.push(id)
+    }
+
+    if (updateProfile) {
+      updateProfile(profile)
+    }
```

Comment:
I think we only want to call this if the update below is successful, so once you move the location of this, I think it should be good to merge.